eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -72,108 +67,93 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
 
   when(WAIT_FOR_PAYMENT_REQUEST) {
     case Event(r: SendMultiPartPayment, _) =>
-      router ! GetNetworkStats
-      goto(WAIT_FOR_NETWORK_STATS) using WaitingForNetworkStats(sender, r)
+      val routeParams = r.getRouteParams(nodeParams, randomize = false) // we don't randomize the first attempt, regardless of configuration choices
```

Comment:
Why not remove the `GetNetworkStats` altogether? Is that really useful?

Why don't we randomize the first attempt?

Reply:
> Why not remove the GetNetworkStats altogether?

Some users told me they're looking at the `NetworkStats` (through the API).
I chose to keep it for now for that reason and because we may want to use it in the future to dynamically choose the values for `min-amount-satoshis` instead of using one provided by users in `eclair.conf`.

> Why don't we randomize the first attempt?

Because it will generate more HTLCs. The non-randomized algorithm is better from an algorithmic point-of-view: it will provide a better set of routes, and a lower total fee so it's preferable. But if these routes fail (because we're blind to remote balances), then it makes sense to introduce randomization in there to try different amounts and routes. It's a heuristic, my guts say it makes sense but we'll have to see it in action!